### ISSUE TITLE
Change default argument `durations` argument from list in `generate_recordings.py`

### ISSUE THREAD

**Issue by JoeZiminski** [SpikeInterface/spikeinterface#2342]
Just leaving here so I don't forget, happy to make a PR on this. Some of the default arguments for functions in `generate_recordings.py` are mutable, which is not reccomended due to [pythons crazy behaviour](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil). These could be changed to `None` and the default value set in the body of the function.

**Comment by h-mayorquin** [comment]
Yeah, I noticed that too recently. This is a common mistake that people make in Python. We should change it. A good heuristic to remember this is that default values are values (not expressions).

**Comment by h-mayorquin** [comment]
This discussion came up again:

https://github.com/SpikeInterface/spikeinterface/pull/2777

 @alejoe91 really wanted to move forward with this, @samuelgarcia supported the idea of using tuples or copying the list at the beginning of the function.
